clients/src/main/java/org/apache/kafka/clients/producer/Partitioner.java

```diff
@@ -44,4 +44,19 @@
      */
     public void close();
 
+    /**
+     * Computes a partition based on relevant data. Knows whether a new batch will be created for the record.
+     * 
+     * @param topic The topic name
+     * @param key The key to partition on (or null if no key)
+     * @param keyBytes The serialized key to partition on( or null if no key)
+     * @param value The value to partition on or null
+     * @param valueBytes The serialized value to partition on or null
+     * @param cluster The current cluster metadata
+     * @param isNewBatch A boolean to specify whether a new batch will be created
+     * @return The record's computed partition
+     */
+    default int computePartition(String topic, Object key, byte[] keyBytes, Object value, byte[] valueBytes, Cluster cluster,  boolean isNewBatch) {
```

Comment:
Could we mark partition() as deprecated and point it to the new method?

Reply:
I marked the method as deprecated in the DefaultPartitioner, but I was not sure about doing the same in the interface. Currently the default (if not overridden) of computePartition is to use the normal partition method. I did this for compatibility reasons--if someone is using a custom partitioner that uses the old partition method. Would deprecating here just be writing a comment above the method in the interface?

Reply:
I was thinking of adding a java deprecated tag.  Let me reply to the voting thread on this so that we can get some consensus on this.

Reply:
So we ended up making some changes and will no longer deprecate the method. Hopefully this will be more clear.